examples/gno.land/p/demo/simpledao/propstore.gno

```diff
@@ -0,0 +1,165 @@
+package simpledao
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/dao"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/demo/ufmt"
+)
+
+var ErrMissingProposal = errors.New("proposal is missing")
+
+const (
+	// maxRequestProposals is the maximum number of
+	// paginated proposals that can be requested
+	maxRequestProposals = 10
+
+	// maxRequestVotes is the maximum number of
+	// paginated votes that can be requested
+	maxRequestVotes = 50
+)
+
+// proposal is the internal simpledao proposal implementation
+type proposal struct {
+	author      std.Address // initiator of the proposal
+	description string      // description of the proposal
+
+	executor dao.Executor       // executor for the proposal
+	status   dao.ProposalStatus // status of the proposal
+
+	tally                 *tally        // voting tally
+	getTotalVotingPowerFn func() uint64 // callback for the total voting power
```

Comment:
i don't get the reason for having this callback.

Reply:
Voting power information is not contained in the proposal, but in the membstore that's associated with the DAO. We need this callback to construct the proposal stats:

```go
func (p *proposal) Stats() dao.Stats {
	// Get the total voting power of the body
	totalPower := p.getTotalVotingPowerFn()

	return dao.Stats{
		YayVotes:         p.tally.yays,
		NayVotes:         p.tally.nays,
		AbstainVotes:     p.tally.abstains,
		TotalVotingPower: totalPower,
	}
}
```